src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/DeleteDataSetVersionFunctionTests.cs

```diff
@@ -0,0 +1,433 @@
+            });
+
+            releaseFile.File.PublicApiDataSetId = dataSet.Id;
+            releaseFile.File.PublicApiDataSetVersion = dataSetVersion.FullSemanticVersion();
+
+            await AddTestData<ContentDbContext>(context => context.Files.Update(releaseFile.File));
+
+            var dataSetVersionDirectory = _dataSetVersionPathResolver.DirectoryPath(dataSetVersion);
+
+            Directory.CreateDirectory(dataSetVersionDirectory);
+            System.IO.File.WriteAllText(Path.Combine(dataSetVersionDirectory, "version1.txt"), "dummy file text");
+
+            await DeleteDataSetVersion(dataSetVersion.Id);
+
+            await using var publicDataDbContext = GetDbContext<PublicDataDbContext>();
+            await using var contentDataDbContext = GetDbContext<ContentDbContext>();
+
+            // Assert that the base directory containing all parquet data file entries has been emptied
+            var dataSetVersionBaseDirectoryEntries = Directory.GetFileSystemEntries(_dataSetVersionPathResolver.BasePath());
+            Assert.Empty(dataSetVersionBaseDirectoryEntries);
+
+            // Assert that the Data Set has been deleted
+            Assert.Null(await publicDataDbContext.DataSets.SingleOrDefaultAsync(ds => ds.Id == dataSet.Id));
+
+            // Assert that the Data Set Version has been deleted
+            Assert.Null(await publicDataDbContext.DataSetVersions.SingleOrDefaultAsync(dsv => dsv.Id == dataSetVersion.Id));
+
+            // Assert that the Data Set Version metadata is deleted
+            Assert.Equal(0,
+                await publicDataDbContext.FilterMetas
+                    .Where(fm => fm.DataSetVersionId == dataSetVersion.Id)
+                    .CountAsync());
+            Assert.Equal(0, 
+                await publicDataDbContext.FilterOptionMetaLinks
+                    .Where(foml => dataSetVersion.FilterMetas.Contains(foml.Meta))
+                    .CountAsync());
+            Assert.Equal(0,
+                await publicDataDbContext.LocationMetas
+                    .Where(fm => fm.DataSetVersionId == dataSetVersion.Id)
+                    .CountAsync());
+            Assert.Equal(0, 
+                await publicDataDbContext.LocationOptionMetaLinks
+                    .Where(loml => dataSetVersion.LocationMetas.Contains(loml.Meta))
+                    .CountAsync());
+            Assert.Equal(0,
+                await publicDataDbContext.IndicatorMetas
+                    .Where(fm => fm.DataSetVersionId == dataSetVersion.Id)
+                    .CountAsync());
+            Assert.Equal(0,
+                await publicDataDbContext.GeographicLevelMetas
+                    .Where(fm => fm.DataSetVersionId == dataSetVersion.Id)
+                    .CountAsync());
+            Assert.Equal(0,
+                await publicDataDbContext.TimePeriodMetas
+                    .Where(fm => fm.DataSetVersionId == dataSetVersion.Id)
+                    .CountAsync());
```

Comment:
Might be a little more concise to use `AnyAsync` here instead, like

```csharp
            Assert.False(await publicDataDbContext.FilterMetas
                    .AnyAsync(fm => fm.DataSetVersionId == dataSetVersion.Id));
```

Reply:
much better!